privacyidea/lib/eventhandler/usernotification.py

```diff
@@ -402,8 +358,9 @@ def do(self, action, options=None):
             recipient = {
                 "email": email
             }
-        else:
-            log.warning("Was not able to determine the recipient for the user "
+
+        elif not action.lower() == "savefile":
```

Comment:
We can just check in `if action == "sendsms":` or `if action == "sendmail":` if the `recipient` is set and issue a warning there.